app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java

```diff
@@ -117,6 +119,7 @@ public class UploadMediaDetailFragment extends UploadBaseFragment implements
      */
     private Place nearbyPlace;
     private UploadItem uploadItem;
+    private LatLng location;
```

Comment:
> location

Better name this one `inAppPictureLocation` for better understanding? Otherwise it is easy to misunderstand this as the value representing actual location in all image upload cases.

Also, a comment clarifying the need to store this separately would also be useful.

A similar adjustment would be needed in other such instances too.

Reply:
>  it is easy to misunderstand this as the value representing actual location in all image upload cases

I intentionally removed this distinction. It's either EXIF location or the current user location. We're treating both of them as _actual._ For all types of uploads, it is anyway the EXIF location that is being used (if available). I thought having this in the initial stages would be enough, it's "picture location" once we decide if it has be used or not. 

I'm sorry for generalising it to this extent, will change it. I have added a detailed comment before `askUserToAllowLocationAccess()`, should I mention it in all the files?